### PR TITLE
Blockchain parameters Foundry tests

### LINKED ISSUE #10793: test/governance/network/blockchainparams.ts

### PR DESCRIPTION
**Pull request by martinvol**
Closes #10793 

Small cleaup of other tests

### CODE REVIEW

File: packages/protocol/test-sol/governance/network/BlockchainParameters.t.Sol [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+
+import "@celo-contracts/governance/BlockchainParameters.sol";
+
+import "celo-foundry/Test.sol";
+
+import { Constants } from "@test-sol/constants.sol";
+import { Utils } from "@test-sol/utils.sol";
+
+contract BlockchainParametersTest is Test, Constants, Utils {
+  uint256 constant EPOCH_SIZE = 100;
```
Comment (soloseng) [inline packages/protocol/test-sol/governance/network/BlockchainParameters.t.Sol]: this is set twice. once in constants, once here

File: packages/protocol/test-sol/identity/Random.t.sol [hunk 2]
```diff
@@ -142,29 +143,29 @@ contract AddTestRandomness is Test {
     // [EPOCH_SIZE, EPOCH_SIZE+1... EPOCH_SIZE+n, 2 * EPOCH_SIZE, 2 * EPOCH_SIZE + 1... 2 * EPOCH_SIZE + RETENTION_WINDOW-1]
 
     // go to last block of epoch 1
+    // TODO add travel here
```
Comment (soloseng) [inline packages/protocol/test-sol/identity/Random.t.sol]: Is this TODO still pending?

File: packages/protocol/test-sol/governance/network/BlockchainParameters.t.Sol [hunk 3]
```diff
@@ -0,0 +1,154 @@
...
+  event BlockGasLimitSet(uint256 limit);
+  event UptimeLookbackWindowSet(uint256 window, uint256 activationEpoch);
+  event OwnershipTransferred(address indexed previousOwner, address indexed newOwner);
+
+  function setUp() public {
+    ph.setEpochSize(EPOCH_SIZE);
+    blockchainParameters = new BlockchainParameters(true);
+  }
+
+  function test_setBlockGasLimit_set_variable() public {
```
Comment (soloseng) [inline packages/protocol/test-sol/governance/network/BlockchainParameters.t.Sol]: Duplicate test

File: packages/protocol/test-sol/governance/network/BlockchainParameters.t.Sol [hunk 4]
```diff
@@ -0,0 +1,154 @@
...
+    assertEq(blockchainParameters.blockGasLimit(), gasLimit);
+  }
+
+  function test_ShoudEmitBlockGasLimitSet () public {
+    vm.expectEmit(true, true, true, true);
+    emit BlockGasLimitSet(gasLimit);
+    blockchainParameters.setBlockGasLimit(gasLimit);
+  }
+
+  function test_ShouldRevert_WhenCalledByNonOwner() public {
```
Comment (soloseng) [inline packages/protocol/test-sol/governance/network/BlockchainParameters.t.Sol]:
```suggestion
  function test_RevertWhen_CalledByNonOwner() public {
```

File: packages/protocol/test-sol/governance/network/BlockchainParameters.t.Sol [hunk 5]
```diff
@@ -0,0 +1,154 @@
...
+    vm.prank(msg.sender);
```
Comment (soloseng) [inline packages/protocol/test-sol/governance/network/BlockchainParameters.t.Sol]: NIT: to keep with the format we have been using, it might be a ggod Idea to create a `nonOwner` account to submit these transactions

File: packages/protocol/test-sol/governance/network/BlockchainParameters.t.Sol [hunk 6]
```diff
@@ -0,0 +1,154 @@
...
+  }
+}
+
+contract BlockchainParametersTest_setIntrinsicGasForAlternativeFeeCurrency is BlockchainParametersTest {
+  function test_ShouldSetTheVariable () public {
+    blockchainParameters.setIntrinsicGasForAlternativeFeeCurrency(gasForNonGoldCurrencies);
+    assertEq(blockchainParameters.intrinsicGasForAlternativeFeeCurrency(), gasForNonGoldCurrencies);
+  }
+
+  function test_ShoudEmitBlockGasLimitSet () public {
```
Comment (soloseng) [inline packages/protocol/test-sol/governance/network/BlockchainParameters.t.Sol]: rename test based on the expected event name

File: packages/protocol/test-sol/governance/network/BlockchainParameters.t.Sol [hunk 7]
```diff
@@ -0,0 +1,154 @@
...
+    vm.expectEmit(true, true, true, true);
+    emit IntrinsicGasForAlternativeFeeCurrencySet(gasForNonGoldCurrencies);
+    blockchainParameters.setIntrinsicGasForAlternativeFeeCurrency(gasForNonGoldCurrencies);
+  }
+
+  function test_ShouldRevert_WhenCalledByNonOwner() public {
```
Comment (soloseng) [inline packages/protocol/test-sol/governance/network/BlockchainParameters.t.Sol]:
```suggestion
  function test_RevertWhen_CalledByNonOwner() public {
```

File: packages/protocol/test-sol/governance/network/BlockchainParameters.t.Sol [hunk 8]
```diff
@@ -0,0 +1,154 @@
...
+    vm.prank(msg.sender);
+    vm.expectRevert("Ownable: caller is not the owner");
+    blockchainParameters.setIntrinsicGasForAlternativeFeeCurrency(gasForNonGoldCurrencies);
+  }
+}
+
+contract BlockchainParametersTest_getUptimeLookbackWindow is BlockchainParametersTest {
+
+  function test_ShouldRevert_WhenNotSet() public {
```
Comment (soloseng) [inline packages/protocol/test-sol/governance/network/BlockchainParameters.t.Sol]: NIT: rename test in an effort to stick to the recommended [best practices](https://book.getfoundry.sh/tutorials/best-practices#general-contract-guidance). Here and throughout

File: packages/protocol/test-sol/governance/network/BlockchainParameters.t.Sol [hunk 9]
```diff
@@ -0,0 +1,154 @@
...
+  }
+
+  function test_MultipleCallsWithinEpochOnlyAppliesLast() public {
+    blockchainParameters.setUptimeLookbackWindow(newValue);
+    blockchainParameters.setUptimeLookbackWindow(otherValue);
+    blockTravel(EPOCH_SIZE);
+    assertEq(blockchainParameters.getUptimeLookbackWindow(), otherValue);
+  }
+
+  function test_ShoudEmitUptimeLookbackWindowSet () public {
```
Comment (soloseng) [inline packages/protocol/test-sol/governance/network/BlockchainParameters.t.Sol]:
```suggestion
  function test_Emit_UptimeLookbackWindowSet () public {
```

File: packages/protocol/test-sol/governance/network/BlockchainParameters.t.Sol [hunk 10]
```diff
@@ -0,0 +1,154 @@
...
+  uint constant lookbackWindow = 20;
+
+  function test_ShouldSetTheVariables() public {
+    blockchainParameters.initialize(gasForNonGoldCurrencies, gasLimit, lookbackWindow);
+    assertEq(blockchainParameters.blockGasLimit(), gasLimit);
+    blockTravel(EPOCH_SIZE);
+    assertEq(blockchainParameters.getUptimeLookbackWindow(), lookbackWindow);
+  }
+
+  function test_ShoudEmitIntrinsicGasForAlternativeFeeCurrencySet() public {
```
Comment (soloseng) [inline packages/protocol/test-sol/governance/network/BlockchainParameters.t.Sol]:
```suggestion
  function test_Emit_IntrinsicGasForAlternativeFeeCurrencySet() public {
```

Comment (soloseng) [inline packages/protocol/test-sol/governance/network/BlockchainParameters.t.Sol]: Might be worth splitting this tet into multiple test, since there are multiple events being emited and the test name is not descriptive of that.

File: packages/protocol/test-sol/governance/network/BlockchainParameters.t.Sol [hunk 11]
```diff
@@ -0,0 +1,154 @@
...
+  }
+
+  function test_setBlockGasLimit_set_variable() public {
+    blockchainParameters.setBlockGasLimit(gasLimit);
+    assertEq(blockchainParameters.blockGasLimit(), gasLimit);
+  }
+
+}
+
+contract BlockchainParametersTest_initialize is BlockchainParametersTest {
```
Comment (soloseng) [inline packages/protocol/test-sol/governance/network/BlockchainParameters.t.Sol]: So far, the conventions has been to use cammel case for the test contract name.


```suggestion
contract BlockchainParametersTestInitialize is BlockchainParametersTest {
```

**Review by soloseng — COMMENTED** [review]
Reviewed. Left a few comments.

Test looks good, but I noticed that the test for this contract are not actually running(local and CI). So it's difficult to tell if they are passing.

File: packages/protocol/test-sol/governance/network/BlockchainParameters.t.Sol (hunk 1 above)
Comment (martinvol) [reply packages/protocol/test-sol/governance/network/BlockchainParameters.t.Sol]: If I use the epoch size in constants there's one edge case it wont be able to test.

**Review by soloseng — CHANGES_REQUESTED** [review]
the test is not actually running

**Review by soloseng — APPROVED** [review]
